packages/gatsby/src/utils/show-experiment-notice.ts

```diff
@@ -35,3 +56,31 @@ export function showExperimentNoticeAfterTimeout(
     clearTimeout(noticeTimeout)
   }
 }
+
+export const createNoticeMessage = (notices): string => {
+  let message = `\nHi from the Gatsby maintainers! Based on what we see in your site, these coming
+features may help you. All of these can be enabled within gatsby-config.js via
+flags (samples below)`
+
+  notices.forEach(
+    notice =>
+      (message += `
+
+${chalk.bgBlue.bold(
+  terminalLink(notice.experimentIdentifier, notice.umbrellaLink)
+)}, ${notice.noticeText}\n`)
+  )
+
+  return message
+}
+
+const showNotices = (): void => {
+  emitter.off(`COMPILATION_DONE`, showNotices)
+  if (noticesToShow.length > 0) {
+    telemetry.trackFeatureIsUsed(`InviteToTryExperiment`)
+    const message = createNoticeMessage(noticesToShow)
+    reporter.info(message)
```

Comment:
Should we also clear `noticesToShow` array once we show message to not show stale messages multiple times (potentially)

Reply:
this should be only called once — if it's called multiple times we probably would want to know to fix that